src/simple-web.js

```diff
@@ -30,37 +30,44 @@ const PASSTHROUGH_CONFIG_KEYS = [
 class SimpleWeb {
 	/**
 	 * @param {SimpleWebServerConfig} config
+	 * @param {http.Server|https.Server} [server]
 	 */
-	constructor(config) {
+	constructor(config, server) {
 		this._config = config;
+		this._server = server;
 
 		this._web = new Koa()
 	}
 
 	start() {
 		return new Promise((resolve, reject) => {
 			if (!this._server) {
-				this._server = this._web.listen(this._config.port);
-
-				PASSTHROUGH_CONFIG_KEYS.forEach((key) => {
-					if (typeof this._config[key] !== 'undefined') {
-						this._server[key] = this._config[key];
-					}
-				});
+				this._createServer();
+			}
+			else if (this._server.listening) {
+				reject(new Error("Server already started"));
 
-				this._server.on("listening", resolve);
+				return;
 			}
 			else {
-				reject(new Error("Server already started"));
+				this._listenOnExistingServer();
 			}
+
+			PASSTHROUGH_CONFIG_KEYS.forEach((key) => {
+				if (typeof this._config[key] !== 'undefined') {
+					this._server[key] = this._config[key];
+				}
+			});
+
+			this._server.on("listening", resolve);
 		});
 	}
 
 	stop() {
 		return new Promise((resolve, reject) => {
-			if (this._server) {
+			if (this._server && this._server.listening) {
 				this._server.on("close", () => {
-					this._server = null;
+					this._server.off("error", reject);
```

Comment:
Where do you ever start listening for the `error` event with the `reject` callback? I don't think this is necessary

Reply:
It's in the existing code at line 68 (now line 75).

Reply:
That will teach me for trying to read the changes on my phone.